cloudinit/util.py

```diff
@@ -2225,11 +2260,18 @@ def write_file(
     @param group: The group to set on the file.
     """
 
-    if preserve_mode:
-        try:
-            mode = get_permissions(filename)
-        except OSError:
-            pass
+    try:
+        fmode = get_permissions(filename)
+    except OSError:
+        fmode = None
+
+    if preserve_mode or (
```

Comment:
I don't think this is the right place to do this. `write_file` is a general purpose utility function, and there may be cases we want to write a new file with completely different (and possibly looser) permissions.

I think `_initialize_filesystem` in `stages.py` is probably the place to do this.

Reply:
@TheRealFalcon fixed in v2.